optimizely/event/user_event_factory.py

```diff
@@ -49,11 +49,12 @@ def create_impression_event(
         if activated_experiment:
             experiment_id = activated_experiment.id
 
-        if variation_id and experiment_id:
-            variation = project_config.get_variation_from_id_by_experiment_id(experiment_id, variation_id)
-        # need this condition when we send events involving forced decisions
-        elif variation_id and flag_key:
+        if variation_id and flag_key:
             variation = project_config.get_flag_variation(flag_key, 'id', variation_id)
+        if not variation and variation_id and experiment_id:
+            # need this condition when we send events involving forced decisions
```

Comment:
```suggestion
            # need this condition when we send events involving forced decisions (F-to-D or E-to-D with any ruleKey/variationKey combinations)
```
Also, Move this comment line to line 52

Reply:
I'm also concerned about line 45-46, which looks like filtering any impression event request with None experiment but FEATURE-TEST source. A valid F-to-D will create such a request for impression event. 
Can you check if we have unit tests and FSC test covering this case - impression event fired for F-to-D?

Reply:
This is a good & critical case, sure I will check it out.

Reply:
We checked it out, and this line won't make it null. 
https://github.com/optimizely/python-sdk/blob/master/optimizely/optimizely.py#L201